zhaquirks/tuya/__init__.py

```diff
@@ -1532,7 +1532,7 @@ class AttributeDefs(BaseAttributeDefs):
         ),
     }
 
-    dp_to_attribute: dict[int, DPToAttributeMapping] = {}
+    dp_to_attribute: dict[int, list[DPToAttributeMapping]] = {}
```

Comment:
Hmm, if we force `dp_to_attribute` to now have a list of `DPToAttributeMapping`, we will break basically all custom Tuya quirks that are floating around. That's still quite a few, since we need to integrate a lot off TRV related quirks (and others) still.

What about allowing `list[DPToAttributeMapping] | DPToAttributeMapping` as a dict value?

We'd need to handle that where `dp_to_attribute` is used OR we could have another attribute of `dict[int, list[DPToAttributeMapping]]` (e.g. `_dp_to_attributes`) and then just convert the `dp_to_attribute` to fit the type of that new attribute (so wrap single `DPToAttributeMapping` in a list). Then use that new attribute in all cases where we process/use `dp_to_attribute` currently.

Reply:
Good point! I forgot that we support custom quirks! I've followed your 2nd approach to make it backwards compatible.